feat: :sparkles: Add Accordion Group Component

elian-dev opened the pull request:
Add Accordion Group Component
- New Styles to the trigger in the _accordion.scss_ file, _also I change the position of some selectors according to the formatter warning._
- New Scripts were added in the _accordions.js_: Two functions were added, one to control the state of a accordion group and the other one to get the state of a group.
- New _Astro file_ to the accordion group component.

**_Pdt: I just add the new component in the design.astro file, but the formatters did their work so other lines were modified._**

eliandev committed:
feat: :sparkles: Add Accordion Group Component

Add Accordion group component

HackMort reviewed (changes requested):
The idea is there for sure, but I would like to take a slightly different approach.

I think is a good idea of always have an accordion wrapper/group, but it should a prop that makes it have the expand/collapse. Here's a video with my idea, please feel free to make any comments of suggestions.


https://github.com/HackMort/jdk-boilerplate/assets/6466646/2d4bc938-09b4-46fd-aba6-f7b61ac8328e

For me the ComponentWrapper should have 2 extra props, 1 to control if the non-active accordions should close when one is active and the other is to have or nor expand/collapse all.

eliandev committed:
fix: :recycle: Refactor the accordions component

HackMort commented on public/assets/js/accordions.js:
```diff
@@ -12,20 +12,57 @@
-        //     accordionItem.setAttribute('aria-expanded', false)
-        //   }
-        // })
+  if (accordions) {
```
We should check here if there's no accordions we should just return the function, is faster and self explanatory.

What you did is fine, is just faster/better do an early return if there's no elements to be manipulated.


```suggestion
  if (!accordions) {
    return;
  }
```

HackMort commented on public/assets/js/accordions.js:
```diff
@@ -12,20 +12,57 @@
+          const expanded = accordionItem.getAttribute('aria-expanded') === 'true' || false
+          accordionItem.setAttribute('aria-expanded', !expanded)
+          // close all other accordion items... maybe?
+          /* accordionItems.forEach((accordionItem) => {
```
Can we double check this still works on a group that can expand all the accordions at once? When I uncommented this code and all the ACC are expanded what it does is close them all first, then I had to click again on the item that I wanted to see and from there it seems to work fine.

HackMort commented on public/assets/js/accordions.js:
```diff
@@ -12,20 +12,57 @@
+          }) */
+
+          if (accordionTrigger) {
+            if (expanded) {
```
We could use a therniary in this statement block, is up to you. When it's one line I'm totally fine with it, but when it's multiple lines o multiple checks in the same therniary is what I'm against it. Agian, it's up to you.

HackMort commented on public/assets/js/accordions.js:
```diff
@@ -12,20 +12,57 @@
+            accordionItem.setAttribute('aria-expanded', false)
+          })
+        } else {
+          accordionTrigger.classList.add('accordion-group-toggle--opened')
```
I see we use the class '.accordion-group-toggle--opened' multiple times, maybe we should have it as variable that we can just change in a single line in the future if we need to.

eliandev committed:
refactor: :recycle: Improve accordions component code

Improve accordions code, modified the function code in order to be more reusable & added documentation to astro the component

eliandev committed:
feat: :art: Add property expand behavior accordions component\

HackMort reviewed (approved):
Everything looks great! Good job!
